ast/compile.go

```diff
@@ -3145,6 +3146,10 @@ func (n *TreeNode) Find(ref Ref) *TreeNode {
 	return node
 }
 
+// Iteratively dereferences ref along the node's subtree.
```

Comment:
Added some docs to explain what's going on here, since it's important for the correctness of the fix above.